lib/remote_server.rb

```diff
@@ -5,6 +5,17 @@ module RemoteServer
   UnknownGitServer = Class.new(RuntimeError)
   UnknownUrlFormat = Class.new(RuntimeError)
   RefDoesNotExist = Class.new(RuntimeError)
+  class AccessDenied < StandardError
+    def initialize(url, action, original_message = nil)
+      @url = url
+      @action = action
+      @original_message = original_message
+    end
+
+    def to_s
+      "Authorization failure when attempting to call #{@url} via #{@action}"
```

Comment:
I considered that but decided against it because the original message does not provide any additional info. It looks like this:
```
body: {"errors":[{"context":null,"message":"You are not permitted to access this resource","exceptionName":"com.atlassian.stash.exception.AuthorisationException"}]}
```